checker/tests/determinism/DefaultField.java

```diff
@@ -0,0 +1,25 @@
+import org.checkerframework.checker.determinism.qual.Det;
+import org.checkerframework.checker.determinism.qual.NonDet;
+import org.checkerframework.checker.determinism.qual.PolyDet;
+import org.checkerframework.framework.qual.HasQualifierParameter;
+
+@HasQualifierParameter(NonDet.class)
```

Comment:
Could you add another test case without `@HasQualifierParameter`? I don't think we have any other tests for field defaults already so now would be a good time to have that case as well.